test/powershell/engine/Remoting/RoleCapabilityFiles.Tests.ps1

```diff
@@ -0,0 +1,94 @@
+        }
+        else
+        {
+            [string] $RoleCapDirectory = (New-Item -Path "$TestDrive\RoleCapability" -ItemType Directory -Force).FullName
+
+            [string] $GoodRoleCapFile = "$RoleCapDirectory\TestGoodRoleCap.psrc"
+            New-PSRoleCapabilityFile -Path $GoodRoleCapFile -VisibleCmdlets 'Get-Command','Get-Process','Clear-Host','Out-Default','Select-Object','Get-FormatData','Get-Help'
+
+            [string] $BadRoleCapFile = "$RoleCapDirectory\TestBadRoleCap.psrc"
+            New-PSRoleCapabilityFile -Path $BadRoleCapFile -VisibleCmdlets *
+            [string] $BadRoleCapFile = $BadRoleCapFile.Replace('.psrc', 'psbad')
+
+            [string] $PSSessionConfigFile = "$RoleCapDirectory\TestConfig.pssc"
+        }
+    }
+
+    AfterAll {
+
+        if (!$IsWindows)
+        {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        }
+    }
+
+    It "Verifies missing role capability file error" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$RoleCapDirectory\NoFile.psrc" }
+        }
+
+        try
+        {
+            $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+            throw 'No Exception!'
+        }
+        catch
+        {
+            ([System.Management.Automation.PSInvalidOperationException] ($_.Exception).InnerException).ErrorRecord.FullyQualifiedErrorId | Should Be 'CouldNotFindRoleCapabilityFile'
+        }
+    }
+
+    It "Verifies incorrect role capability file extenstion error" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$BadRoleCapFile" }
+        }
+
+        try
+        {
+            $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+            throw 'No Exception!'
+        }
+        catch
+        {
+            ([System.Management.Automation.PSInvalidOperationException] ($_.Exception).InnerException).ErrorRecord.FullyQualifiedErrorId | Should Be 'InvalidRoleCapabilityFileExtension'
+        }
+    }
+
+    It "Verifies restriction on good role capability file" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$GoodRoleCapFile" }
+        }
+
+        # 'Get-Service' is not included in the session.
+        $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+        [powershell] $ps = [powershell]::Create($iss)
+        $null = $ps.AddCommand('Get-Service')
+
+        try
+        {
+            $ps.Invoke()
+            throw 'No Exception!'
+        }
+        catch
+        {
+            ($_.Exception).InnerException.GetType().FullName | Should Be 'System.Management.Automation.CommandNotFoundException'
```

Comment:
The pattern we can use for all test above too.
Only here `GetType()` call may raise a "null reference" exception.
```powershell
        try
        {
            $ps.Invoke()
            throw 'No Exception!'
        }
        catch
        {
            if ($_.Exception.InnerException) {
                $exc_name = $_.Exception.InnerException.GetType().FullName 
            }
            $exc_name | Should Be 'System.Management.Automation.CommandNotFoundException'
        }
```